### PR TITLE
Pass --host_action_env to host options hostActionEnvironment attribute

### PR DESCRIPTION
**Pull request by mai93**
This PR passes the value of `host_action_env` to `hostActionEnvironment` attribute of the host options in `CoreOptions.getHost`. 

Fixes: #12403

### CODE REVIEW

File: src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java [hunk 1]
```diff
@@ -109,8 +109,9 @@ public boolean isHostTransition() {
 
     @Override
     public BuildOptions patch(BuildOptionsView options, EventHandler eventHandler) {
-      if (executionPlatform == null) {
-        // No execution platform is known, so don't change anything.
+      if (executionPlatform == null || options.get(CoreOptions.class).isExec) {
```
Comment (katre) [inline src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java:112]: This isn't safe: it's entirely possible to need to build tools in order to build other tools. How does this lead to the error with action_env?

Comment (mai93) [reply src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java:112]: I thought that the error of conflicting actions may be caused by having two copies of the same execution transition. Adding this check fixed the --host_action_env conflicting actions issue, so I created this PR to just to make sure it does not break other tests.

Comment (katre) [reply src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java:112]: If the two configurations were exactly the same, they would have the same hash. Can you check what the difference between the two is? I do not think this change is safe to submit.

You can diff two configurations by using the "bazel config" command, with the same flags, after you've run the command to cause the error. The config subcommand checks for configurations present in skyframe's cache, so anything that invalidates the cache will make this fail.

Comment (mai93) [reply src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java:112]: The only difference between the 2 configurations is com.google.devtools.build.lib.analysis.config.CoreOptions.action_env where one config includes the env var passed with host_action_env and the other doesn't.

Comment (katre) [reply src/main/java/com/google/devtools/build/lib/analysis/config/ExecutionTransitionFactory.java:112]: I see that the value of `hostActionEnvironment` isn't being kept in `CoreOptions.getHost`. If you add the line, does that deal with the issue?

```
    // Pass host action environment variables
    host.actionEnvironment = hostActionEnvironment;
    host.hostActionEnvironment = hostActionEnvironment;
```

**Comment by katre** [comment]
Can you add a test for the host_action_env issue? It looks like there aren't any tests for the flag at all, putting them in https://cs.opensource.google/bazel/bazel/+/master:src/test/shell/integration/action_env_test.sh?ss=bazel&q=action_env would probably be appropriate.

File: .bazelci/postsubmit.yml [hunk 2]
```diff
@@ -257,16 +257,52 @@ tasks:
       - test
   rbe_ubuntu1604:
     shell_commands:
-      - sed -i.bak -e 's/^# android_sdk_repository/android_sdk_repository/'
-        -e 's/^# android_ndk_repository/android_ndk_repository/'
-        -e 's/^# rbe_autoconfig/rbe_autoconfig/'
-        -e 's/^# load("@bazel_toolchains/load("@bazel_toolchains/' WORKSPACE
+      - sed -i.bak
```
Comment (katre) [inline .bazelci/postsubmit.yml:260]: Why are there changes in the postsubmit and presubmit? We should get these reviewed by someone on who knows them better than I do.

Comment (mai93) [reply .bazelci/postsubmit.yml:260]: I included these commits to fix the failing presubmit jdeps_test as mentioned in #12685

Comment (katre) [reply .bazelci/postsubmit.yml:260]: They should be submitted (and reviewed) separately, since it's not directly part of fixing this issue.

Comment (mai93) [reply .bazelci/postsubmit.yml:260]: They are already submitted, rebasing the pull request should fix this, right?

Comment (katre) [reply .bazelci/postsubmit.yml:260]: Oh, I see. Yes, rebase onto master and that should be fine.

File: src/test/shell/integration/action_env_test.sh [hunk 3]
```diff
@@ -218,4 +239,205 @@ function test_action_env_changes_honored {
...
+
+function test_host_env_using_successive_exec_tools_simple() {
+  export FOO=baz
+
+  bazel build --host_action_env=FOO=bar pkg:show_host_env_using_successive_exec_tools \
+      || fail "${PRODUCT_NAME} build show_host_env_using_successive_exec_tool failed"
+
+  cat `bazel info ${PRODUCT_NAME}-genfiles`/pkg/successive_exec_tools_env.txt > $TEST_log
+  expect_log "FOO=bar"
+
```
Comment (katre) [inline src/test/shell/integration/action_env_test.sh:440]: Remove this extra blank line.

**Review by katre — APPROVED** [review]
One small nit, but otherwise this looks great. Thanks for all the work on this.

**Pull request closed without merging**